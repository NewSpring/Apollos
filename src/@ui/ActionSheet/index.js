export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
