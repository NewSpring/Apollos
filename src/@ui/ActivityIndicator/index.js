import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
