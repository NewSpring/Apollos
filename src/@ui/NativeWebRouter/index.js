export { default as Router } from './Router';
export { default as Link, goBackTo } from './Link';
export { default as AndroidBackButton } from './AndroidBackButton';
export { default as matchLocationToPath } from './matchLocationToPath';
export { default as ProtectedRoute } from './ProtectedRoute';
export { default as withProtectedFunction } from './withProtectedFunction';
export { default as DeepLinking } from './DeepLinking';

export { Route, Switch, withRouter, Redirect, matchPath } from 'react-router';
