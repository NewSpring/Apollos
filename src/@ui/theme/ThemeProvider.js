import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import createTheme, { THEME_PROPS } from './createTheme';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    themeInput: {},
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: createTheme(this.props.themeInput),
    themeInput: this.props.themeInput,
  })

  render() {
    return Children.only(this.props.children);
  }
}
