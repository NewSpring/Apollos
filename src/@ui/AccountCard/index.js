import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
