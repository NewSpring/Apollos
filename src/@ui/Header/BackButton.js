import React from 'react';
import { View, Platform } from 'react-native';
import { compose, withProps } from 'recompose';
import MediaQuery from '@ui/MediaQuery';
import { Link } from '@ui/NativeWebRouter';
import { H7 } from '@ui/typography';
import Icon from '@ui/Icon';
import styled from '@ui/styled';

const Row = styled({ flexDirection: 'row', alignItems: 'center', justifyContent: 'center' })(View);

const Container = styled(
  ({ theme }) => ({
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
    zIndex: 5, // fixes hitSlop by positiong button above header text
    alignItems: 'center',
    justifyContent: 'center',
    paddingLeft: 4,
    paddingRight: 100, // adds extra hitSlop space.
    ...Platform.select({
      web: {
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        left: theme.sizing.baseUnit,
        top: theme.sizing.baseUnit,
      },
    }),
  }),
  'BackButton.Container',
)(View);

const LargerTappableLink = compose(
  withProps({
    /* Mostly arbitrary padding/hitSlop values taken from padding values above
     * `SafeAreaView` doesn't appear to like hitSlop so padding is used on iOS.
     */
    style: Platform.OS === 'ios' ? { paddingRight: 100, paddingLeft: 4 } : null,
    hitSlop: Platform.OS === 'android' ? { right: 100, left: 4 } : null,
  }),
)(Link);

const BackButton = props => (
  <Container>
    <LargerTappableLink pop>
      <Row>
        <Icon name="arrow-back" size={24} {...props} />
        <MediaQuery minWidth="md">
          <H7>Back</H7>
        </MediaQuery>
      </Row>
    </LargerTappableLink>
  </Container>
);

export default BackButton;
