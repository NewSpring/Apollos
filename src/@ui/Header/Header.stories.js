import React from 'react';

import { storiesOf } from '@storybook/react-native';

import Header from '@ui/Header';

storiesOf('Header', module)
  .add('renders', () => (
    <Header titleText="Welcome!" />
  ));

