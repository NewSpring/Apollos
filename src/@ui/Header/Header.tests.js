import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@ui/ThemeProvider';
import { Router } from '@ui/NativeWebRouter';

import Header from './';

describe('The Header component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Header />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a back button correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Router>
          <Header backButton />
        </Router>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
