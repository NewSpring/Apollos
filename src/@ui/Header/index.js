import React from 'react';
import { View, StatusBar, Platform } from 'react-native';
import SafeAreaView from '@ui/SafeAreaView';
import PropTypes from 'prop-types';
import { compose, setPropTypes, pure, branch, renderNothing, defaultProps } from 'recompose';
import { withTheme, withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';
import { H2, H6 } from '@ui/typography';

import BackButton from './BackButton';

const StyledHeaderBar = styled(
  ({ theme }) => ({
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 25,
    paddingHorizontal: theme.sizing.baseUnit / 2,
    ...Platform.select({
      web: {
        height: undefined,
        paddingHorizontal: theme.sizing.baseUnit,
        paddingVertical: theme.sizing.baseUnit,
        paddingTop: theme.sizing.baseUnit * 2.5,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'flex-start',
      },
    }),
  }),
  'Header.Bar',
)(View);

const HeaderContainer = styled(
  ({ theme, backgroundColor }) => ({
    backgroundColor: backgroundColor || theme.colors.background.primary,
    ...Platform.select({
      android: {
        // paddingTop: 25, // todo: this is currently required as SafeAreaView isn't
        // properly adding padding on android.
      },
      web: {
        backgroundColor: theme.colors.background.paper,
      },
    }),
  }),
  'Header.Container',
)(SafeAreaView);

const WebHeaderText = styled(
  ({ theme }) => ({
    color: theme.colors.darkSecondary,
  }),
  'Header.WebHeaderText',
)(H2);

const StyledHeaderText =
  Platform.OS === 'web'
    ? WebHeaderText
    : styled(
      ({ theme, barStyle }) => ({
        color: barStyle === 'dark-content' ? theme.colors.darkPrimary : theme.colors.lightPrimary,
        maxWidth: '80%',
      }),
      'Header.Text',
    )(H6);

const RightContainer = styled(
  ({ theme }) => ({
    position: 'absolute',
    right: 4,
    top: 0,
    bottom: 0,
    justifyContent: 'center',
    ...Platform.select({
      web: {
        right: theme.sizing.baseUnit,
        top: theme.sizing.baseUnit,
        justifyContent: 'flex-start',
      },
    }),
  }),
  'Header.RightContainer',
)(View);

const ColoredBackButton = compose(
  pure,
  withTheme(({ theme, barstyle }) => ({
    color: barstyle === 'dark-content' ? theme.colors.darkPrimary : undefined,
  })),
)(BackButton);

const enhance = compose(
  defaultProps({
    backButton: false,
    barStyle: 'light-content',
    children: null,
  }),
  setPropTypes({
    webEnabled: PropTypes.bool,
    titleText: PropTypes.string,
    backButton: PropTypes.bool,
    barStyle: PropTypes.oneOf(['light-content', 'dark-content']),
    children: PropTypes.node,
  }),
  branch(({ webEnabled }) => !webEnabled && Platform.OS === 'web', renderNothing),
  branch(
    () => Platform.OS !== 'web',
    withThemeMixin(({ theme, barStyle }) => ({
      type: barStyle === 'light-content' ? 'dark' : 'light',
      colors: {
        background: {
          ...theme.colors.background,
          default: theme.colors.background.primary,
        },
      },
    })),
  ),
  pure,
);

const Header = enhance(
  ({
    titleText,
    right,
    backButton = false,
    barStyle = 'light-content',
    style = {},
    backgroundColor = null,
    children,
  }) => (
    <HeaderContainer backgroundColor={backgroundColor} style={style}>
      <StatusBar barStyle={barStyle} />
      <StyledHeaderBar>
        {backButton ? <ColoredBackButton barStyle={barStyle} /> : null}
        {titleText ? (
          <StyledHeaderText barStyle={barStyle} numberOfLines={1}>
            {titleText}
          </StyledHeaderText>
        ) : null}
        {children}
        {right ? <RightContainer>{right}</RightContainer> : null}
      </StyledHeaderBar>
    </HeaderContainer>
  ),
);

export default Header;
