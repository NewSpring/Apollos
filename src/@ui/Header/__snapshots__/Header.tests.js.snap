// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component should render correctly 1`] = `
<RCTSafeAreaView
  backgroundColor={null}
  style={
    Object {
      "backgroundColor": "#6bac43",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`The Header component should render with a back button correctly 1`] = `
<RCTSafeAreaView
  backgroundColor={null}
  style={
    Object {
      "backgroundColor": "#6bac43",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 4,
          "paddingRight": 100,
          "position": "absolute",
          "top": 0,
          "zIndex": 5,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={null}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingLeft": 4,
            "paddingRight": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Svg
            barStyle="light-content"
            color={undefined}
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <Path
              d="M14.35 21L6.2 12.46c-.26-.28-.26-.64 0-.92L14.34 3l1.65.9L8.27 12 16 20.1l-1.65.9z"
              fill="#ffffff"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
